### ISSUE TITLE
DETR: Training From Scratch not Learning Anything

### ISSUE THREAD

**Issue by PatelKrutarth** [NielsRogge/Transformers-Tutorials#139]
Hello,

Thank you so much for creating all the demo/tutorial notebooks for huggingface models!

I am trying to fine-tune and train the DETR model from scratch to increase the number of detection objects. Finetuning works fine. However, when I do training from scratch by loading a model using a config instead of from_pretrained() method, the training process goes without any error, however, the model is not learning anything and the loss remains almost the same. I have tried for 200-300 epochs to train with lr=1e-4, lr_backbone=1e-5.

My labeled data does not have image segmentation labels, I have only object detection labels.

Below is the snippet to load the model:

```
         if from_scratch:
            configuration = DetrConfig(num_queries=max_objects, num_labels=num_labels)
            self.model = DetrForObjectDetection(configuration)
         else:
            self.model = DetrForObjectDetection.from_pretrained("facebook/detr-resnet-50", 
                                                                num_labels=num_labels,
                                                                ignore_mismatched_sizes=True)
```

I am not sure what I am doing wrong here. Please let me know how I can train the DETR model from scratch.

**Comment by r-remus** [comment]
I'm experiencing the same issue; I'm also trying to train DETR from scratch for the same reason (i.e., to increase the number of object queries).

I'm init'ing the model as follows:
```
            config = DetrConfig(
                backbone='resnet50',
                id2label=id2label,
                label2id=label2id,
                num_labels=len(label2id),
                num_queries=250,
                use_pretrained_backbone=True,
            )
            return DetrForObjectDetection(config=config)
```

**Comment by sounakdey** [comment]
I am experiencing the same issue @PatelKrutarth and @r-remus ... did you find a work around
